stdlib/src/builtin/string.mojo

```diff
@@ -2100,3 +2174,169 @@ fn _calc_format_buffer_size[type: DType]() -> Int:
         return 64 + 1
     else:
         return 128 + 1  # Add 1 for the terminator
+
+
+# ===----------------------------------------------------------------------===#
+# Format method structures
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct _FormatCurlyEntry:
+    """
+    Internally used by the `format()` method.
+
+    Specifically to structure fields.
+
+    Does not contains any substitution values.
+
+    """
+
+    var first_curly: Int
+    """the index of an opening brace around a substitution field"""
+
+    var last_curly: Int
+    """the index of an closing brace around a substitution field"""
+
+    var field: Variant[
+        String,  # kwargs indexing (`{field_name}`)
+        Int,  # args manual indexing (`{3}`)
+        NoneType,  # args automatic indexing (`{}`)
+        Bool,  # for escaped curlies ('{{')
+    ]
+    """Store the substitution field."""
+
+    fn _is_escaped_brace(ref [_]self) -> Bool:
+        return self.field.isa[Bool]()
```

Comment:
nit: here and below: since the struct itself is already hidden, there is no need to make its methods hidden too, especially since they are used outside the struct itself.
```suggestion
    fn is_escaped_brace(ref [_]self) -> Bool:
        return self.field.isa[Bool]()
```